test/e2e/e2e_test.go

```diff
@@ -893,13 +955,20 @@ func TestE2E(t *testing.T) {
 		}
 	})
 
-	// Test that the operator's metrics endpoint is being read by prometheus
+	promClient, err := newPrometheusClientForConfig(adminConfig)
+	if err != nil {
+		t.Fatalf("error initializing prometheus client: %v", err)
+	}
 	t.Run("metrics", func(t *testing.T) {
-		promClient, err := newPrometheusClientForConfig(adminConfig)
```

Comment:
Please move the client initialization back to the test so that it will only run if the test does.

Reply:
I was trying to follow this test pattern mentioned in the golang blog where setup code is before the test https://blog.golang.org/subtests. Look for the section `Setup and Tear-down`

    func TestTeardownParallel(t *testing.T) {
        // <setup code>
        // This Run will not return until its parallel subtests complete.
        t.Run("group", func(t *testing.T) {
            t.Run("Test1", parallelTest1)
            t.Run("Test2", parallelTest2)
            t.Run("Test3", parallelTest3)
        })
        // <tear-down code>
    }

Do you still want me to move the initialization back to the test?

Reply:
Where setup code is used by all tests it should appear outside of it. Since the setup is only used by the metrics tests, it can appear inside of the wrapping test before the subtests are defined e.g.

```golang
func TestFoo(...) {

  t.Run("metrics", func(...) {
    <setup>
    t.Run("metrics test 1", func(...) {}
    t.Run("metrics test 2", func(...) {}
  }
}
```

Note that we're not enabling parallel execution of these tests. Care would have to be taken to do so because the rotation tests aren't compatible with concurrent execution - they affect global state that other tests depend on.

Reply:
updated the code and moved intialization inside `t.Run`